### ISSUE TITLE
Inquiries about the activitynet dataset

### ISSUE THREAD

**Issue by 18162195164** [bpiyush/TestOfTime#8]
Hello author, regarding the questions raised by others before, you replied with an Excel file about the training set, evaluation set, and validation set of activitynet. However, I found that the training set does not have the field "annotation_id", while both the evaluation and test sets have this field. Could you please provide me with a corresponding Excel file? Thank you!

**Comment by bpiyush** [comment]
Hi. Thanks for reaching out. I am not sure which previous file you are referring to. Could you please elaborate what exactly do you need?

**Comment by 18162195164** [comment]
非常感谢您的回复，不好意思我之前好像没有表达清楚，我在之前您回复其他人的问题中下载了您提供的activitynet数据集的分割文件，但是当我使用的时候发现在他的评估集和测试集中有annotation_id这个字段，但是在训练集中没有annotation_id这个字段，我想询问一下这个字段是如何生成的，如何做到精准匹配。

**Comment by 18162195164** [comment]
Thank you very much for your reply. I'm sorry that I didn't express myself clearly before. I downloaded the split file of the activitynet dataset you provided when you replied to others' questions before. However, when I used it, I found that there was the field "annotation_id" in its evaluation set and test set. However, there is no "annotation_id" field in the training set. I would like to ask how this field is generated and how to achieve precise matching.

**Comment by 18162195164** [comment]
Thank you very much for your reply. Sorry, I didn't express my meaning clearly before. I downloaded the split file of the activitynet dataset that you provided when answering others' questions before. However, when I used it, I found that there was a field "annotation_id" in its val set and test set. However, there is no "annotation_id" field in the training set. I would like to ask how this field is generated and how exact matching is achieved. Could you tell me how to generate this field or give me your corresponding file?

**Comment by bpiyush** [comment]
Hi @18162195164 ,

Apologies again for a delayed response. I was busy with a deadline.

I had forgotten what `annotation_id` is since it has been a long time. But I found the code which generated it.

* It is nothing but a ID of the clip within the broader video. So if a video `v_NLdyQ1oMmAo` has 10 clips, then we assign an ID to these clips as `v_NLdyQ1oMmAo-1`, `v_NLdyQ1oMmAo-2`, .., `v_NLdyQ1oMmAo-10`.
* Below, I am providing code that assigns this `annotation_id`.

```python
data_dir = "/.../datasets/activitynet-1.3/"
mode = "val_1"

# Load split file
split_path = join(data_dir, f"annotations/captions/{mode}-temporal_v2.0.csv")
assert exists(split_path)
data = pd.read_csv(split_path)


def add_counter_suffix_to_column(df, col, new_col=None):
    new_col = col + "_counter" if new_col is None else new_col
    g = df.groupby(col)
    df[new_col] = df[col] + g.cumcount().add(1).astype(str).radd('_').mask(
        g[col].transform('count')==1, '',
    )
    return df

data = add_counter_suffix_to_column(data, "video_id", "annotation_id")
```

I hope this helps.

**Comment by 18162195164** [comment]
Hello, Piyush Bagad! Thank you for your reply.
Regarding the activitynet dataset, in the split you shared in the issue before, I found that the validation set and the test set only have the two fields of "video" and "annotation_id". I would like to ask if these two files are complete, how you handled this dataset, and could you give me your activitynet.py file?

**Comment by 18162195164** [comment]
Hello, Piyush Bagad! Thank you for your reply.
Regarding the activitynet dataset, in the split you shared in the issue before, I found that the validation set and the test set only have the two fields of "video" and "annotation_id". I would like to ask if these two files are complete and if any other complete data files are missing: activitynet_val.json and activitynet_test.json?

**Comment by bpiyush** [comment]
I realise that I probably made an error while saving the public val/test split files for activitynet. Unfortunately, I did this work during my MSc and I no longer have access to the old split files I have used.

May be a workaround is that you can use the method described in the paper to create a new split? Or do you want to benchmark against the exact same split?

**Comment by 18162195164** [comment]
Hello, under the file activitynet_val_temporal_1k.csv you provided, you named video,annotation_id
v_-02DygXbn6w,v_-02DygXbn6w_6
v_-02DygXbn6w,v_-02DygXbn6w_12. Then we found the file activitynet_captions_val1.json, and the corresponding part is {
"video_id": "v_-02DygXbn6w",
"video": "v_-02DygXbn6w.mkv",
"caption":  "Two people are very slowly snow boarding down a mountain of snow. They fall over for a minute and get back up,  the one farther behind catches up and gets in front of the other boarder. The boarder tried to grab on to a sign and is  falling in and tumbling around in snow. Seems like it was a bit of a struggle to finally get to the bottom.",
"source": "ActivityNet_Captions",
"duration": 109.71000000000001
"timestamps": [
[
0,
19.75
],
[
19.75,
53.21
],
[
53.21,
80.08
],
[
80.08,
109.71
]
],
"sentences": [
"Two people are very slowly snow boarding down a mountain of snow.",
"They fall over for a minute and get back up,  the one farther behind catches up and gets in front of the other boarder.",
"The boarder tried to grab on to a sign and is falling in and tumbling around in snow.",
"Seems like it was a bit of a struggle to finally get to the bottom."
]
},
I would like to ask which specific timing statement v_-02DygXbn6w_6 is and by what rules do you evaluate the sequence. How can I obtain all the timing statements corresponding to the annotation_id through this sample?
